glide-core/src/client/value_conversion.rs

```diff
@@ -673,6 +674,289 @@ pub(crate) fn convert_to_expected_type(
+                        return Err((ErrorKind::TypeError, "Incorrect value type received for group value. Wanted an Array").into());
+                    };
+
+                    let consumers_key = Value::SimpleString("consumers".into());
+                    let opt_consumers_index = group
+                        .iter()
+                        .position(
+                            |key| {
+                                let res = convert_to_expected_type(key.clone(), Some(ExpectedReturnType::SimpleString));
+                                match res {
+                                    Ok(converted_key) => {
+                                        converted_key == consumers_key
+                                    },
+                                    Err(_) => {
+                                        false
+                                    }
+                                }
+                            }
+                        );
+
+                    let Some(consumers_index) = opt_consumers_index else {
+                        return Err((ErrorKind::TypeError, "No consumers key found").into());
+                    };
+
+                    if group.get(consumers_index + 1).is_none() {
+                        return Err((ErrorKind::TypeError, "No consumers value found.").into());
+                    }
+
+                    let Value::Array(ref consumers) = group[consumers_index + 1] else {
+                        return Err((ErrorKind::TypeError, "Incorrect value type received for consumers. Wanted an Array.").into());
+                    };
+
+                    if consumers.is_empty() {
+                        groups_as_maps.push(
+                            convert_to_expected_type(Value::Array(group.clone()), Some(ExpectedReturnType::Map {
+                                key_type: &Some(ExpectedReturnType::BulkString),
+                                value_type: &None,
+                            }))?
+                        );
+                        continue;
+                    }
+
+                    let mut consumers_as_maps = Vec::new();
+                    for consumer in consumers {
+                        consumers_as_maps.push(convert_to_expected_type(consumer.clone(), Some(ExpectedReturnType::Map {
+                            key_type: &Some(ExpectedReturnType::BulkString),
+                            value_type: &None,
+                        }))?);
+                    }
+
+                    group[consumers_index + 1] = Value::Array(consumers_as_maps);
+                    let group_map = convert_to_expected_type(Value::Array(group), Some(ExpectedReturnType::Map {
+                        key_type: &Some(ExpectedReturnType::BulkString),
+                        value_type: &None,
+                    }))?;
+                    groups_as_maps.push(group_map);
+                }
+
+                array[groups_index + 1] = Value::Array(groups_as_maps);
```

Comment:
Refactor the index so we don't keep having to add 1 to it. Same for other similar indices.